tools/raisinbread_refresh.php

```diff
@@ -0,0 +1,138 @@
+<?php declare(strict_types=1);
+/**
+ * This script is intended for developers working with test data. It DROPs all
+ * LORIS core tables as well as "Raisinbread" test instrument tables.
+ * For this reason, it should obviously not be run on any server wil live data.
+ *
+ * After dropping the tables, it will source the Raisinbread test data using
+ * the commands found in raisinbread/README.md.
+ *
+ * The script also restores the url and host config settings to their pre-drop
+ * values. This prevents an issue where a developer will need to manually
+ * change these values in a MySQL shell when they are not hosting a LORIS
+ * on localhost.
+ *
+ * Finally, the script runs tools/resetpassword.php so that the default admin
+ * password with the Raisinbread data set is not used.
+ *
+ * In order to prevent accidental data loss, the script prompts the user to
+ * manually type the exact name of the database which will be affected. It
+ * also requires the user to manually enter their database password before
+ * dropping and importing.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Loris
+ * @author   John Saigle <john.saigle@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+
+$info = <<<INFO
+This script is used by LORIS developers to empty their databases and replace
+them with new test data.
+DO NOT RUN THIS SCRIPT ON PRODUCTION.
+
+
+INFO;
+
+echo $info;
+
+try {
+    require_once 'generic_includes.php';
+
+    $dbInfo = $config->getSettingFromXML('database');
+    $dbname = $dbInfo['database'];
+    $host = $dbInfo['host'];
+    $username = $dbInfo['username'];
+
+    $urlConfigSetting = $config->getSetting('url');
+    $hostConfigSetting = $config->getSetting('host');
+
+    echo 'Please enter the name of your database to continue.' . PHP_EOL;
```

Comment:
I think your TRY block is too wide here, it feels like you are adding it only for lines 50-51.

Reply:
That's what I did at first but `getSettingFromXML()`will crash with a PHP Fatal Error if the right tables don't exist. That happened to me when I had a different error after dropping but before importing that crashed the script. 

Might be an edge-case but I figure there's no harm making it wide.

Reply:
oh i see.. it's fine then

Reply:
I improved the exception handling so that things are more fine-grained.